pritunl-ssh: init at 1.0.1674.4

Thunderbottom opened the pull request:
pritunl-ssh is a small Python script used to retrieve the SSH certificate from the Pritunl Zero server.

Homepage: https://zero.pritunl.com/
GitHub: https://github.com/pritunl/pritunl-zero-client

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

rmcgibbo commented on maintainers/maintainer-list.nix:
```diff
@@ -9363,6 +9363,16 @@
     githubId = 1391883;
     name = "Tom Hall";
   };
+  Thunderbottom = {
```
Can you split this into a separate commit with a message like 'maintainers: add Thunderbottom'

Thunderbottom replied:
this is done, thanks for spotting!

rmcgibbo commented on pkgs/tools/networking/pritunl-ssh/default.nix:
```diff
@@ -0,0 +1,31 @@
+    sha256 = "07z60lipbwm0p7s2bxcij21jid8w4nyh6xk2qq5qdm4acq4k1i88";
+  };
+
+  patchPhase = ''
```
Can you try using `patchShebangs` instead? This should be handled semi-automatically.

Thunderbottom replied:
I am not sure if I am getting this wrong, or maybe it seems to not be working for me for some reason.

I've tried adding `patchShebangs ssh_client.py` to the `patchPhase`, and that seems to not be working (i.e. does not output anything during the `nix-build`). I've also tried replacing `patchPhase` with `postPatch`, containing:
```
  postPatch = ''
    patchShebangs ssh_client.py
    patchShebangs ssh_host_client.py
  '';
```

which at least outputs the following during the `nix-build`:

```
...
patching script interpreter paths in ssh_client.py
patching script interpreter paths in ssh_host_client.py
...
```

But when I check the binary in `result/bin`, it still has `#!/usr/bin/env python3` at the top.

Am I doing something wrong?

SuperSandro2000 replied:
I think you need to add python3 to build inputs.

Thunderbottom replied:
that worked, thank you!

Thunderbottom committed:
pritunl-ssh: init at 1.0.1674.4

pritunl-ssh is a small Python script used to retrieve
the SSH certificate from the Pritunl Zero server.

Homepage: zero.pritunl.com
GitHub: pritunl/pritunl-zero-client

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

Thunderbottom committed:
maintainers: add Thunderbottom

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

SuperSandro2000 commented on pkgs/tools/networking/pritunl-ssh/default.nix:
```diff
@@ -0,0 +1,31 @@
+  src = fetchFromGitHub {
+    owner = "pritunl";
+    repo = "pritunl-zero-client";
+    rev = "${version}";
```
```suggestion
    rev = version;
```

Thunderbottom replied:
fixed, thanks for spotting!

SuperSandro2000 commented:
Please squash all the package related commits together.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113112` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>pritunl-ssh</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

pritunl-ssh:
<details>


`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/networking/pritunl-ssh/default.nix:16:3:

```
   |
16 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
